migrator/migrator-servlet/src/main/java/com/google/phonenumbers/ServletMain.java

```diff
@@ -48,8 +61,26 @@ protected void doPost(HttpServletRequest req, HttpServletResponse resp) throws S
       }
       handleSingleNumberMigration(number, countryCode, req, resp);
     } else {
-      String file = req.getParameter("file");
-      countryCode = req.getParameter("fileCountryCode");
+      handleFileMigration(req, resp);
+    }
+  }
+
+  /** Retrieves the form data for the numbers after a file migration and downloads them as a text file. */
+  @Override
+  protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
+    String fileName = "+" + req.getParameter("countryCode") + "_Migration_ " + req.getParameter("fileName");
```

Comment:
I believe we should do at least some sanitization on those request parameters before building a filename from them, e. g. exclude certain special characters like `/`, `\`, `.` etc. (not an exhaustive list). Can you check if there's a library for that? :)

Reply:
+1